pkg/plugin/driver/daemonset/daemonset_test.go

```diff
@@ -166,3 +171,153 @@ func TestFillTemplate(t *testing.T) {
 		t.Errorf("Expected annotations key1:val1 and key2:val2 to be set, but got %v", daemonSet.Spec.Template.Annotations)
 	}
 }
+
+func TestMonitorOnce(t *testing.T) {
+	// Note: the pods/ds must be marked with the label "sonobuoy-run" or else our labelSelector
+	// logic will filter them out even though the fake server returns them.
+
+	// We will need to be able to grab these items repeatedly and tweak the minor details
+	// so these helpers make the test cases much more readable.
+	testPlugin := &Plugin{Base: driver.Base{Definition: plugin.Definition{Name: "myPlugin"}}}
+	validDS := appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"sonobuoy-run": ""}},
+	}
+	validPod := func(node string) corev1.Pod {
+		return corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"sonobuoy-run": ""}},
+			Spec:       corev1.PodSpec{NodeName: node},
+		}
+	}
+	failingPod := func(node string) corev1.Pod {
+		return corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"sonobuoy-run": ""}},
+			Spec:       corev1.PodSpec{NodeName: node},
+			Status: corev1.PodStatus{
+				Conditions: []corev1.PodCondition{
+					{Reason: "Unschedulable", Message: "conditionMsg"},
```

Comment:
It might make the error message checks below a bit clear/explicit if we could include the node name in the condition message because then we know which error message corresponds to which node. Just a suggestion though.

Reply:
Can you make an issue for that and we can make that change separately? It is rare to see that error condition at all but I agree with you it could be clarified.